### PR TITLE
Clarifies VID:PID requirements

### PR DESCRIPTION
**Pull request by JellyTitan**
Adds:
- VID:PID's do not need to be unique, but is recommended
- `0xFEED` as VID is not required, and is not allowed in VIA
- Adds note about pid.codes, and pid.codes is not required

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/faq_build.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ You can buy a really unique VID:PID here. I don't think you need this for person
 - https://www.obdev.at/products/vusb/license.html
 - https://www.mcselec.com/index.php?page=shop.product_details&flypage=shop.flypage&product_id=92&option=com_phpshop&Itemid=1
 
+Open-source hardware projects can request a unique [USB-IF](https://www.usb.org/) recognized USB VID:PID by submitting a pull request to [pid.codes](https://pid.codes/). Pid.codes will all alwyas use `0x1209` for the VID. A unique PID is selected when submitting a pull request to pid.codes. Using pid.codes for your VID:PID is not required, and is probably not needed for personal use.
```
Comment (drashna) [inline docs/faq_build.md:63]: pid.codes is one site that does this, but Microchip offers the same for their controllers, V-USB has a range too, IIRC.  

* pid.codes
* https://microchip.my.site.com/s/article/How-to-use-a-USB-VID-and-PID-from-the-USB-sub-licensing-program
* https://wiki.openmoko.org/wiki/USB_Product_IDs

And I think there is another group out there that has a license that allows for sub-licensing, too

File: docs/faq_build.md [hunk 2]
```diff
@@ -49,9 +49,9 @@ Re-running the QMK installation script (`./util/qmk_install.sh` from the `qmk_fi
 If that doesn't work, then you may need to download and run Zadig. See [Bootloader Driver Installation with Zadig](driver_installation_zadig.md) for more detailed information.
 
 ## USB VID and PID
-You can use any ID you want with editing `config.h`. Using any presumably unused ID will be no problem in fact except for very low chance of collision with other product.
+You can use any ID you want with editing `config.h`. Using an ID that is presumed to be unused should generally not be a problem, except for the possibility of a rare collision with another product. A unique PID/VID combination is not required, but it is recommended.
```
Comment (drashna) [inline docs/faq_build.md:52]: Might be worth noting that the 0x3000 block is being actively assigned, and getting pretty far in the range.  If you're avoiding collisions, the further past 0x4000, the better.

File: docs/faq_build.md (hunk 1 above)
Comment (JellyTitan) [reply docs/faq_build.md]: Thanks! I've updated the docs with notes about the other vendors you listed. I wasn't able to find anything regarding USB-V - but I'm a novice and may be looking in the wrong place.

File: docs/faq_build.md (hunk 2 above)
Comment (JellyTitan) [reply docs/faq_build.md]: @drashna 
I updated the docs with your note about avoiding collisions above 0x4000. 
I'm a bit confused though - I thought the VID/PID codes correspond to integers in hexadecimal format and not memory blocks - so there wouldn't necessarily be ranges? Maybe I'm missing something?

I first encountered the VID/PID when trying to setup VIA, and my first reference there was the [scotto keeb](https://www.youtube.com/watch?v=7d5yzBOup9U&themeRefresh=1) video, which presents the VID/PID as 2 characters converted to hexadecimal. I suspect many other folks will start there too.

What would be the best way to approach the VID/PID conceptually for people new to it? Maybe something like:
> Vendor ID (VID) and Product ID (PID) codes are unique identifiers assigned to USB devices by the USB Implementers Forum (USB-IF) to distinguish them from other USB devices. These codes are used by the USB host to identify the type and manufacturer of the USB device. VID and PID are always represented in hexadecimal,  prefix  written in hexadecimal format with a "0x". QMK does not require a unique PID/VID combination.

Comment (drashna) [reply docs/faq_build.md]: The IDs are arbitrary and don't actually represent anything.  But the USB-IF (eg, the regulatory body that controls the USB specs, vendor ID assignments, etc) hand them out in a sequential manner.  (there are gaps and such in lower values, but for the 0x3xxx block, it's been very sequential).  

Anything like scotto's stuff is just fluff around that, and an easy way to pick a VID/PID. I've seen sha hashs of the keyboard/manu's name being used for generating it too.  

So really, there isn't any real association. It's just a number that is assigned, and used to identify the hardware.

**Pull request closed without merging**